### DISCUSSION TITLE
OSM Tagging validation

### DISCUSSION THREAD

**Discussion by Cj-Malone** [alltheplaces/alltheplaces#9352] [Ideas]
As part of https://github.com/alltheplaces/alltheplaces/issues/5598#issue-1830590248 I've started trying to compare the output against [id-tagging-schema](https://github.com/openstreetmap/id-tagging-schema). No (good) code yet, but it's on my TODO list.

Most of the differences are fine, if our extra attributes are valid OSM we should try and improve id-tagging-schema, or ignore it. If we have an attribute that doesn't exist in OSM, we should try and find an alternative, or ignore it.

If we collide with an existing OSM tag, we should fix that.

The big ones that we should ignore are `addr:street_address` and `located_id(:wikidata)?`. As already discussed, they aren't OSM, but the former is the best we can (reasonably) do, and the latter has no OSM equivalent, yet.

The one big one that has the potential to just be noise is, `contact:email`. It is valid OSM, but id-tagging-schema considers `email` the better tag. I'm no longer fighting the `contact` namespace fight, I'm happy (and want to) change to `email`. It could be considered a breaking change, but honestly, who is consuming our email addresses? And it get's us closer to OSM, so it should be easier for people to consume after. `contact` for social media has won, we don't need to change that.

Fuels and fuel card things come up again, but is nothing special. We may be able to document some things on the OSM wiki.

**Comment by brawer** [comment]
Could we send the values of `addr:street_address` to a geocoding API such as [OpenCage](https://opencagedata.com/) or Nominatim? Given the potential value of ATP for OSM, a commercial geocoding company such as OpenCage might be open to giving us an API key for free, if we ask nicely. Also, interfacing with a geocoder would eventually allow us to crawl stores that plain publish address lists without coordinates.

**Comment by Cj-Malone** [comment]
I have played with geocoding before, it was way to slow to be usable on a large scale. But presumably the network was the slowest bit, so local Nominatim might be better.

https://github.com/alltheplaces/alltheplaces/pull/5595#issuecomment-1657264450

**Comment by jimbofreedman** [comment]
Did you mean `located_in`? I haven't seen `located_id` yet.

Also, I'm intending to try out local Nominatim a bit further down our pipe sometime soon - I can let you know how it goes.

**Comment by CloCkWeRX** [comment]
Is it worth moving this one to discussions?
Or is there a way to define "done" on it? 
Right now it feels fairly open ended.

**Reply by iandees** [reply]
I moved it to a discussion.
